### PR TITLE
Use short array syntax; add return types; require return types

### PR DESCRIPTION
**Pull request by uuf6429**

### CODE REVIEW

File: tests/Basic/BestPracticesTest.php [hunk 1]
```diff
@@ -2,26 +2,25 @@
...
 final class BestPracticesTest extends TestCase
 {
-    public function testExtendsCoreDriver()
+    public function testExtendsCoreDriver(): void
     {
         $driver = $this->createDriver();
 
-        $this->assertInstanceOf('Behat\Mink\Driver\CoreDriver', $driver);
-
-        return $driver;
```
Comment (uuf6429) [inline tests/Basic/BestPracticesTest.php:18]: No idea why this test was returning the driver...so I removed it, hope that's fine?

File: tests/Basic/ErrorHandlingTest.php [hunk 2]
```diff
@@ -9,13 +9,13 @@
...
 final class ErrorHandlingTest extends TestCase
 {
-    const NOT_FOUND_XPATH = '//html/./invalid';
+    private const NOT_FOUND_XPATH = '//html/./invalid';
 
-    const NOT_FOUND_EXCEPTION = 'Exception';
+    private const NOT_FOUND_EXCEPTION = 'Exception';
 
-    const INVALID_EXCEPTION = 'Exception';
+    private const INVALID_EXCEPTION = 'Exception';
```
Comment (uuf6429) [inline tests/Basic/ErrorHandlingTest.php:16]: These are technically a BC break, but I think the risk of that is very low.

If we don't want to risk it, we can use `@internal`/`@deprecated` + `public` instead.

**Review by stof — APPROVED** [review]

File: tests/TestCase.php [hunk 3]
```diff
@@ -41,7 +41,7 @@ public static function prepareSession()
      *
      * @throws \UnexpectedValueException if the global driver_config_factory returns an invalid object
      */
-    private static function getConfig()
+    private static function getConfig(): AbstractConfig
```
Comment (uuf6429) [inline tests/TestCase.php:44]: I figured that changing the signature of a private method should be BC safe enough,

File: tests/Basic/ErrorHandlingTest.php (hunk 2 above)
Comment (stof) [reply tests/Basic/ErrorHandlingTest.php:16]: these constants have no reason to be reused by driver repositories, so this is fine.

**Pull request merged**